components/db/testimony/usePublishedTestimonyListing.ts

```diff
@@ -40,21 +48,26 @@ export type TestimonyFilterOptions =
 export type UsePublishedTestimonyListing = ReturnType<
   typeof usePublishedTestimonyListing
 >
+
 export function usePublishedTestimonyListing({
   uid,
-  billId
+  billId,
+  published,
 }: {
   uid?: string
   billId?: string
+  published?: string
```

Comment:
we should remove the published option here since it's implied that this hook returns published testimony. It looks like this was added to support both draft and published testimony on the testimony page? In that case, We can create a new hook useBillTestimony that contains the getBillTestimony logic and use it on the testimony page.

Right now, if you call getBillTestimony to view someone else's testimony, it will throw a permission denied error since it can't access the draft document. useBillTestimony can handle this by only fetching the draft if the authenticated user matches the requested uid. That way the testimony page rendering logic is the same in both cases: call useBillTestimony, show the published version if draft.publishedVersion is set, otherwise show the draft if it exists, otherwise 404.

I don't think anything else currently uses getBillTestimony so feel free to change it/absorb it into the hook.